cpp/tests/centrality/edge_betweenness_centrality_test.cu

```diff
@@ -0,0 +1,340 @@
+{
+  result_t rescale_factor            = static_cast<result_t>(1);
+  result_t casted_number_of_vertices = static_cast<result_t>(number_of_vertices);
+  if (normalize) {
+    if (number_of_vertices > 1) {
+      rescale_factor /= ((casted_number_of_vertices) * (casted_number_of_vertices - 1));
+    }
+  } else {
+    if (!directed) { rescale_factor /= static_cast<result_t>(2); }
+  }
+  for (auto idx = 0; idx < number_of_edges; ++idx) { result[idx] *= rescale_factor; }
+}
+
+template <typename VT, typename ET, typename WT, typename result_t>
+void reference_edge_betweenness_centrality(
+  cugraph::experimental::GraphCSRView<VT, ET, WT> const &graph,
+  result_t *result,
+  bool normalize,
+  VT const number_of_sources,
+  VT const *sources)
+{
+  VT number_of_vertices = graph.number_of_vertices;
+  ET number_of_edges    = graph.number_of_edges;
+  thrust::host_vector<VT> h_indices(number_of_edges);
+  thrust::host_vector<ET> h_offsets(number_of_vertices + 1);
+
+  thrust::device_ptr<VT> d_indices((VT *)&graph.indices[0]);
+  thrust::device_ptr<ET> d_offsets((ET *)&graph.offsets[0]);
+
+  thrust::copy(d_indices, d_indices + number_of_edges, h_indices.begin());
+  thrust::copy(d_offsets, d_offsets + (number_of_vertices + 1), h_offsets.begin());
+
+  cudaDeviceSynchronize();
+
+  reference_edge_betweenness_centrality_impl<VT, ET, WT, result_t>(
+    &h_indices[0], &h_offsets[0], number_of_vertices, result, sources, number_of_sources);
+  reference_rescale<VT, ET, WT, result_t>(
+    result, graph.prop.directed, normalize, number_of_vertices, number_of_edges);
+}
+// Explicit declaration
+template void reference_edge_betweenness_centrality<int, int, float, float>(
+  cugraph::experimental::GraphCSRView<int, int, float> const &,
+  float *,
+  bool,
+  const int,
+  int const *);
+template void reference_edge_betweenness_centrality<int, int, double, double>(
+  cugraph::experimental::GraphCSRView<int, int, double> const &,
+  double *,
+  bool,
+  const int,
+  int const *);
+
+// =============================================================================
+// Utility functions
+// =============================================================================
+// Compare while allowing relatie error of epsilon
+// zero_threshold indicates when  we should drop comparison for small numbers
+template <typename T, typename precision_t>
+bool compare_close(const T &a, const T &b, const precision_t epsilon, precision_t zero_threshold)
+{
+  return ((zero_threshold > a && zero_threshold > b)) ||
+         (a >= b * (1.0 - epsilon)) && (a <= b * (1.0 + epsilon));
+}
+
+// =============================================================================
+// Test Suite
+// =============================================================================
+// Defines Betweenness Centrality UseCase
+// SSSP's test suite code uses type of Graph parameter that could be used
+// (MTX / RMAT)
+// FIXME: Use VT for number_of_sources?
```

Comment:
I doubt anything other than int for VT is supported in the backend.

Reply:
Removed the comment in 7e7b668.